### PR TITLE
ensure versions for output 'blas' don't overlap

### PR DESCRIPTION
**Pull request by h-vetinari**
The versions strings for output `blas` were colliding with the ones that had been produced for lapack 3.8.0 (at least). Disambiguate by bumping the number that goes into those version strings.

### CODE REVIEW

**Comment by h-vetinari** [comment]
@isuruf 

After #59, `conda create -n test python numpy "blas=*=blis"` does not work as expected, and I believe the version numbering of the `blas`-output is the issue. The above command pulls in:
```
The following NEW packages will be INSTALLED:

  blas               conda-forge/win-64::blas-2.3-blis
  blis               conda-forge/win-64::blis-0.7.0-h62dcd97_1
  [...]
  libblas            conda-forge/win-64::libblas-3.9.0-3_blis
  libcblas           conda-forge/win-64::libcblas-3.9.0-3_blis
  liblapack          conda-forge/win-64::liblapack-3.9.0-3_hd5c7e75_netlib
  liblapacke         conda-forge/win-64::liblapacke-3.9.0-3_hd5c7e75_netlib
```

On the other hand, this works `conda create -n test python numpy "libblas=*=5_blis"`, which pulls in:
```
The following NEW packages will be INSTALLED:

  blis               conda-forge/win-64::blis-0.8.0-h8d14728_0
  [...]
  libblas            conda-forge/win-64::libblas-3.9.0-5_blis
  libcblas           conda-forge/win-64::libcblas-3.9.0-5_blis
  liblapack          conda-forge/win-64::liblapack-3.9.0-3_hd5c7e75_netlib
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -2,6 +2,10 @@
 {% set build_num = 5 %}
```
Comment (isuruf) [inline recipe/meta.yaml:2]: Let's change this to `20` and make a comment that it should not be reset to 0

Comment (h-vetinari) [reply recipe/meta.yaml:2]: It's not by a big margin, but I don't think that's a better solution, for two reasons:
* new lapack versions _should_ IMO reset the build number (aside from being confusing for CF-regulars, the GH PR template invites to), and there's really two different constructs being versioned here (lapack & CF-blas-arch), which don't _have to_ have the same `build_num`
* in case there's ever a need to create a branch for rebuilding for an older lapack-version, it would be very intransparent what the right `build_num` should be

Comment (isuruf) [reply recipe/meta.yaml:2]: I don't really agree with 1., but 2 is good enough. Can you add a comment above that if it is reset, then blas_minor offset needs to be updated.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: Thanks, done.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -220,6 +224,7 @@ outputs:
         - if not exist %LIBRARY_BIN%/liblapacke.dll exit 1           # [win]
 
   - name: blas-devel
+    # uses lapack {{ version }}, not {{ blas_major }}
```
Comment (h-vetinari) [inline recipe/meta.yaml:227]: I was double-checking this, because it's not immediately clear from the recipe what's happening here. To be honest, I was surprised that `blas-devel` is using a different versioning scheme than `blas`, but whether intentional or not, it probably can't be fixed or changed until `blas_major` overtakes the lapack version. In the meantime, I thought it's better to leave a comment.

File: recipe/meta.yaml (hunk 1 above)
Comment (h-vetinari) [reply recipe/meta.yaml:3]: As a matter of fact, what do you think about creating a 3.8 branch here and we do one more build that has a consistent build number & corresponding blas-version? Because currently, there's a bit of an eclectic [mix](https://anaconda.org/conda-forge/blas/files) with blas builds for 3.9 but with low numbers

Comment (isuruf) [reply recipe/meta.yaml:3]: what does that achieve? builds with highest build number are 3.8 right now.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: Some of the low numbers (like `osx-arm64/blas-2.5-openblas.tar.bz2`) actually are for 3.9, and it's basically build-dependent whether it's 3.8 or 3.9 (for example a new `win-64/blas-2.2-openblas.tar.bz2` was built 2 months ago - I only know that it cannot be 3.9 because #55 hadn't been merged yet, but then why does it have such a low build_num?)

Comment (h-vetinari) [reply recipe/meta.yaml:3]: So the idea would be to have a definite max build_num across 3.8 builds on all platforms, so that the resolver doesn't have to chase down rabbit holes based on the apparent patchwork of conditions (that lead to lower build_nums being more current).

Comment (isuruf) [reply recipe/meta.yaml:3]: But what would creating a 3.8 branch achieve?

Comment (isuruf) [reply recipe/meta.yaml:3]: max build_num across 3.8 builds is 21 on all platforms. what am I missing?

Comment (h-vetinari) [reply recipe/meta.yaml:3]: With such a branch, we could have another 3.8 build `libblas-3.8.0-22_{blis,mkl,openblas}` that _uniformly_ corresponds to a `blas-2.22_{blis,mkl,openblas}`, without having to switch back and forth for the needs of 3.9/master.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: > max build_num across 3.8 builds is 21 on all platforms. what am I missing?

That the low `blas` build_nums (now `blas_minor`) got overwritten once lapack moved to 3.9.0, and now there's a mix of `blas` builds that either belong to 3.8/3.9, depending on whether the old package had been deleted/retired/etc.

e.g. `win-64/blas-2.2-openblas.tar.bz2` should be ancient, but it is one of the most recent [builds](https://anaconda.org/conda-forge/blas/files)

Comment (isuruf) [reply recipe/meta.yaml:3]: Ah, I see what you are saying, but I'd like to avoid unnecessary builds. If you want, please send a PR to mark the 3.9.0 builds with low numbers as broken.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: OK cool, glad we're on the same page.

However, compared to identifying which packages exactly were wrongly overwritten, marking them broken (breaking the reproducibility of people's environments, and possible breaking lapack-3.8-only deployments completely), creating a branch and running CI for 2h seems like the much better trade-off?!

Comment (isuruf) [reply recipe/meta.yaml:3]: I'm sorry, but I still don't see any advantage for a rebuild. Right now the ones with the highest build numbers are 3.8 and doing a rebuild wouldn't change this. (The only thing that would change is that they'll have the same build number.)

Comment (h-vetinari) [reply recipe/meta.yaml:3]: Yes, the ones with the highest number are 3.8, but the highest number is often _not the most up-to-date_ 3.8 build (leading to outdated requirements e.g. with compilers etc.)! Again the example `win-64/blas-2.2-openblas.tar.bz2` (is definitely a 3.8 build, as it came before #55), which is the second-most-recent windows+openblas build...

Just have a look at the `blas`-[files](https://anaconda.org/conda-forge/blas/files) and see the "chaos" for yourself.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: Ah wait, sorry, now I confused myself with 3.9 vs MKL builds...

Comment (isuruf) [reply recipe/meta.yaml:3]: Nope, `blas-2.2` is a 3.9 build.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: I realised... see updated comment above.

Comment (isuruf) [reply recipe/meta.yaml:3]: It'll still be hard to follow as well. `netlib` builds are done with a different build number as well and I don't think we'll be able to keep the build numbers in sync for different blas implementations and IMO that's okay.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: Sure, but the `netlib` ones don't have an associated `blas` package.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: Nevermind, they do. OK, I'll drop this, thanks for the exchange

Comment (isuruf) [reply recipe/meta.yaml:3]: Thanks for the discussion. I really appreciate your work on better BLAS support on conda-forge.

Comment (h-vetinari) [reply recipe/meta.yaml:3]: Happy to hear it! :)
